autopilot/hostfilter.go

```diff
@@ -0,0 +1,152 @@
+package autopilot
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"net"
+
+	"go.sia.tech/renterd/bus"
+	"go.sia.tech/siad/modules"
+	"go.sia.tech/siad/types"
+)
+
+const (
+	// minContractFundUploadThreshold is the percentage of contract funds
+	// remaining at which the contract gets marked as not good for upload
+	minContractFundUploadThreshold = float64(0.05) // 5%
+)
+
+type hostFilter struct {
+	host     host
+	metadata bus.ContractMetadata
+	reasons  []string
+
+	gfu bool
+	gfr bool
```

Comment:
I'd call those initial* or something. I initially thought either these or the metadata are redundant until I realised this is used for creating the "reasons"